common/lib/capa/capa/templates/matlabinput.html

```diff
@@ -82,27 +81,34 @@
           answer = input.serialize();
 
           // a chain of callbacks, each querying the server on success of the previous one
-
-          var get_callback = function(response) {
-              var new_result_elem = $(response.html).find(".ungraded-matlab-result").html();
-              var external_grader_msg = $(response.html).find(".external-grader-message").html();
-              result_elem = $(problem_elt).find(".ungraded-matlab-result");
-              result_elem.addClass("is-fading-in");
-              result_elem.html(new_result_elem);
-              external_grader_msg_elem = $(problem_elt).find(".external-grader-message");
-              external_grader_msg_elem.addClass("is-fading-in");
-              external_grader_msg_elem.html(external_grader_msg);
-              if (!external_grader_msg.trim()) {
-                  matlab_result_task.task_poller.stop();
-              } else {
-                  result_elem.html('');
-              }
+          var poll = function(prev_timeout) {
+              $.postWithPrefix(url + "/problem_get", function(response) {
+                   var new_result_elem = $(response.html).find(".ungraded-matlab-result").html();
+                   var external_grader_msg = $(response.html).find(".external-grader-message").html();
+                   result_elem = $(problem_elt).find(".ungraded-matlab-result");
+                   result_elem.addClass("is-fading-in");
+                   result_elem.html(new_result_elem);
+                   external_grader_msg_elem = $(problem_elt).find(".external-grader-message");
+                   external_grader_msg_elem.addClass("is-fading-in");
+                   external_grader_msg_elem.html(external_grader_msg);
+                   // If we have a message about waiting for the external grader.
+                   if(external_grader_msg.trim()) {
+                       result_elem.html('');
+                        // Setup the polling for the next round
+                        var next_timeout = prev_timeout * 2;
+                        // The XML parsing that capa uses doesn't handle the greater-than symbol properly here, so we are forced to work around it.
+                        if(next_timeout - 64000 == 0) {
```

Comment:
Because of exponential growth, it will go from 1000, 2000,...,32000, 64000. Because of the fact that capa is terrible, it needs to be exact. We can't use `>` or `<` here.